### PR TITLE
Added compatibility for flarum-tag-password extension

### PR DESCRIPTION
**Pull request by Yippy**
Add support for [flarum-tag-passwords](https://github.com/datlechin/flarum-tag-passwords), where Admin can lock specific tag with Password and Group Permission, the template would need to prevent loading the stats and last discussion.

Here is an example of it turned on:
image

image

### CODE REVIEW

**Comment by Yippy** [comment]
Fix the background colour for issue [askvortsov1/flarum-categories#41] [askvortsov1/flarum-categories#45]

image

image

**Comment by Yippy** [comment]
Fix design when view is very narrow, put spacing to ensure icon is not grouped closely to text.

image

**Comment by Yippy** [comment]
Fixed: Missing 'Tags' in navigation, with 'Keep the tags page link in the nav sidebar?'
Fixed: Compact mobile mode view design

image

File: js/src/forum/index.js [hunk 1]
```diff
@@ -9,10 +9,19 @@ import LastDiscussionWidget from './components/LastDiscussionWidget';
 import StatWidget from './components/StatWidget';
 
 function pruneIndexNav(items, func) {
-  if (app.current.matches(CategoriesPage) && app.forum.attribute('categories.fullPageDesktop')) {
+  const isTagsVisible = !app.forum.attribute('categories.keep-tags-nav');
```
Comment (askvortsov1) [inline js/src/forum/index.js]: Why is this negated?

File: js/src/forum/components/Category.tsx [hunk 2]
```diff
@@ -17,11 +17,18 @@ interface Attrs {
   parent: any;
 }
 
+interface TagLocked {
```
Comment (askvortsov1) [inline js/src/forum/components/Category.tsx:26]: I'm not completely opposed to including features for the tag lock extension here, but would it be possible to add a comment here explaining which extension is adding this?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Hey, thank you so much for these fixes and contributions, really sorry it took so long to get to. Left a few minor comments but otherwise looks good!

File: js/src/forum/index.js (hunk 1 above)
Comment (Yippy) [reply js/src/forum/index.js]: Removing the tag navigation should not be checked if fullPageDesktop is true, hence why it's removed.  This caused the Tag navigation to be shown for the user.

File: js/src/forum/components/Category.tsx (hunk 2 above)
Comment (Yippy) [reply js/src/forum/components/Category.tsx:26]: Added comment for reference of Flarum Tag Passwords

File: js/src/forum/index.js (hunk 1 above)
Comment (askvortsov1) [reply js/src/forum/index.js]: Ah, I agree about the fullPageDesktop change. But you have `isTagsVisible` being set to the opposite of whether the user wants to keep tags. Shouldn't it be:

`isTagsVisible = app.forum.attribute(... keep-tags-nav)`?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: js/src/forum/index.js (hunk 1 above)
Comment (Yippy) [reply js/src/forum/index.js]: It seems I've used the wrong attributes, the latest commit is 'categories.keepTagsNav' and fixed the logic

**Review by askvortsov1 — APPROVED** [review]
Thanks again for putting this together!

**Pull request merged**